gateway/dataimpl/server_v1/errorhandler.go

```diff
@@ -275,6 +275,57 @@ func (e ErrorHandler) NewQueryIndexNotBuildingStatus(baseErr error, bucketName,
 	return st
 }
 
+func (e ErrorHandler) NewQueryIndexScopeOrCollectionNotFoundStatus(
```

Comment:
Why did you choose to create a unique type of error creation here rather than reusing the existing NewCollectionNotFoundStatus and NewScopeNotFoundStatus methods?  I noticed the main difference is likely that this function generates errors with a ResourceType of 'queryindex' rather than 'scope' or 'collection'.

Reply:
Honestly I didn't see those statuses but I  was thinking do we not want to have the query index name in the error status we're returning since whatever's using STG will want to know what index was being created when the error occurred?